### PR TITLE
Make enemies find a wrapping path to the player if it is shorter

### PR DESCRIPTION
**Pull request by kalabukdima**
I think this makes the game a bit harder. But it's totally fine if you don't like this and decline PR.

### CODE REVIEW

File: src/geometry/point.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ impl Point {
     }
 
     /// Returns the squared distance from this point to the given one
-    pub fn squared_distance_to(&self, target: &Point) -> f32 {
+    pub fn squared_distance_to(self, target: Point) -> f32 {
```
Comment (kalabukdima) [inline src/geometry/point.rs:29]: [Clippy](https://github.com/rust-lang-nursery/rust-clippy) stated: "_this argument is passed by reference, but would be more efficient if passed by value_".

File: src/geometry/size.rs [hunk 2]
```diff
@@ -19,7 +19,7 @@ impl Size {
     }
 
     /// Returns true if the `Point` is contained in this `Size` or false otherwise
-    pub fn contains(&self, point: Point) -> bool {
+    pub fn _contains(&self, point: Point) -> bool {
```
Comment (aochagavia) [inline src/geometry/size.rs]: Why the underscore? Is the function no longer in use?

**Comment by aochagavia** [comment]
Thanks for the PR, it looks great!

File: src/geometry/size.rs (hunk 2 above)
Comment (kalabukdima) [reply src/geometry/size.rs]: Yes, it is no longer in use. But it looks useful. Should I just remove it?

**Comment by kalabukdima** [comment]
Should I worry about failed checks? Seems that tests have already been broken.

File: src/geometry/size.rs (hunk 2 above)
Comment (aochagavia) [reply src/geometry/size.rs]: Yes, we can always rescue it from the git history if necessary

**Comment by aochagavia** [comment]
Checks have been failing since I switched to nightly. I just pushed a PR so Appveyor knows we are using nightly now, so in the future checks should succeed

**Comment by aochagavia** [comment]
Great, thanks!

**Pull request merged**